File: sephora-frontend/src/models/user/GetUserDto.ts
import RatingDto from "../rating/RatingDto.ts";
import OrderDto from "../order/OrderDto.ts";
import PictureDto from "../picture/PictureDto.ts";
import CartItem from "../Cart/CartItem.ts";

export default interface GetUserDto {
    get id(): string;

    get firstName(): string;

    get lastName(): string;

    get email(): string;

    get profilePicture(): PictureDto | null;

    get phoneNumber(): string | null;

    get roles(): string[];

    get registrationDate(): Date;

    get ratings(): RatingDto[];

    get orders(): OrderDto[];

    get cartItems(): CartItem[];
}

File: sephora-backend/Application/Dtos/User/GetUserDTO.cs
namespace CleanArchitecture.Application.Dtos.User;

public class GetUserDto
{
    public string Id { get; set; } = String.Empty;

    public string FirstName { get; set; } = String.Empty;

    public string LastName { get; set; } = String.Empty;

    public string Email { get; set; } = String.Empty;

    public PictureDto? ProfilePicture { get; set; }

    public string? PhoneNumber { get; set; }

    public List<string> Roles { get; set; } = [];

    public DateTime RegistrationDate { get; set; }

    public ICollection<RatingDto> Ratings { get; set; } = [];

    public ICollection<OrderDto> Orders { get; set; } = [];
    
    public ICollection<CartDto> CartItems { get; set; } = [];
}
